### PR TITLE
Makes flash hyper vulnerability based on negative flash resistance modifier, removes snowflake trait.

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request
There was a line of code in flashes that checked for a specific trait to determine if someone was flashable from all sides that was completely independent from the previously established eye protection framework. The trait was only implemented in two places in the entire game, Removes all traces of the TRAIT_FLASH_SENSITIVE, replaces the check with one that will detect negative flash protection. Achievable by flash vulnerable eyewear, and mutant eyes.

Now, giving a pair of sunglasses to a moth would now put their eyes on par with normal unprotected eyes. Where as before, there would be no difference between a moth with sunglasses, and a moth without.

## Why It's Good For The Game
The current interaction is weird and doesn't make much sense. This ties in a standalone mechanic with a preexisting and developed system.

## Changelog

:cl: itseasytosee
code: Being able to be flashed from all sides is now based on or not you have a negative flash resistance, rather than randomly applying to specific pairs of eyes.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
relevant: https://github.com/tgstation/tgstation/pull/55350#issuecomment-739060523

**Comment by Mothblocks** [comment]
i think your pr isn't *just* #55350 but i already denied "moths can get flashed from every direction" so you might want to rethink this

**Comment by itseasytosee** [comment]
> i think your pr isn't _just_ #55350 but i already denied "moths can get flashed from every direction" so you might want to rethink this

I understand your conclusion.
As a proposed compromise, require that to be flashed in all directions a carbon has to have a eye protection of -2 or less. Fly and maintenance eyes will be given FLASH_PROTECTION_HYPER_SENSITIVE defined as a -2 modifier. Night vison goggles will remain as -1, meaning that moths who wear night vison goggles or similar equipment will be flash from all directions.
Do you have thoughts on this?

**Comment by Mothblocks** [comment]
sounds good

**Comment by itseasytosee** [comment]
Changes implemented

**Comment by JohnFulpWillard** [comment]
When I added flypeople flashing in https://github.com/tgstation/tgstation/pull/55724 I mentioned that a large reason I wanted this was because Flypeople realistically do see everything around them because of their eyes.

Ironically, realistically sunglasses would be protecting only the direction they're facing and they'd be left vulnerable from everywhere else, but no one really cares about flypeople anyways so it likely doesn't even matter.

**Comment by itseasytosee** [comment]
> When I added flypeople flashing in #55724 I mentioned that a large reason I wanted this was because Flypeople realistically do see everything around them because of their eyes.
> 
> Ironically, realistically sunglasses would be protecting only the direction they're facing and they'd be left vulnerable from everywhere else, but no one really cares about flypeople anyways so it likely doesn't even matter.

The solution to this complicated balancing issue is to add vison cones.

File: code/modules/assembly/flash.dm [hunk 1]
```diff
@@ -200,8 +200,10 @@
 	if(victim.flags_1 & IS_SPINNING_1)
 		return DEVIATION_NONE
 
-	if(HAS_TRAIT(victim, TRAIT_FLASH_SENSITIVE)) //If your eyes are sensitive and can be flashed from any direction.
-		return DEVIATION_NONE
+	if(iscarbon(victim))
+		var/mob/living/carbon/carbon_victim = victim
+		if(carbon_victim.get_eye_protection() < -1) // If we have really bad flash sensitivity, usually due to really sensitive eyes, we get flashed from all directions
```
Comment (Mothblocks) [inline code/modules/assembly/flash.dm:205]:
```suggestion
		if(carbon_victim.get_eye_protection() < FLASH_PROTECTION_SENSITIVE ) // If we have really bad flash sensitivity, usually due to really sensitive eyes, we get flashed from all directions
```

**Comment by Mothblocks** [comment]
don't redefine the macro, put it in _DEFINES

**Comment by itseasytosee** [comment]
> don't redefine the macro, put it in _DEFINES

You right g

**Pull request merged**